### ISSUE TITLE
In 'from' clause, allow 'variable = value'

### ISSUE THREAD

**Issue by julianhyde** [julianhyde/morel#11]
In 'from' clause, allow 'variable = value'. For example,
```
from e in hr.emps,
 fullName = e.firstName ^ " " ^ e.lastName
yield {e.empno, fullName};
```

It is syntactic sugar for iterating over a singleton collection:
```
from e in hr.emps,
 fullName in [e.firstName ^ " " ^ e.lastName]
yield {e.empno, fullName};
```

But I think the syntax may be useful for users.

In this case you could rewrite to a `let`:
```
from e in hr.emps
yield
  let
    val fullName = e.firstName ^ " " ^ e.lastName
  in
    {e.empno, fullName}
  end;
```
But this would not be possible if `fullName` were used in a later `in` or a `where` clause.

Further, now that #62 has implement the function <tt>only: &alpha; list &rarr; &alpha;</tt> (which throws if the list is empty or has more than one element), we can write
```
from e in hr.emps,
  d = only (from d in hr.depts where d.deptno = e.deptno)
yield
  {e, d};
```
This is important in asserting that a sub-query returns precisely one row. The same effect as a scalar sub-query in SQL:
```
SELECT e.*,
  (SELECT d.name
    FROM hr.depts AS d
    WHERE d.deptno = e.deptno) AS dname
FROM hr.emps AS e
```
(SQL is a little more limited, because you are only allowed to return 1 column.)

**Comment by julianhyde** [comment]
Note the similarity with the 'let' clause in XQuery ('let' is the 'L' in 'FLWOR'). Here is example 6-6 in [XQuery by Priscilla Walmsley](http://www.datypic.com/books/xquery/chapter06.html):
```
let $doc := doc("catalog.xml")
for $prod in $doc//product
let $prodDept := $prod/@dept
let $prodName := $prod/name
where $prodDept = "ACC" or $prodDept = "WMN"
return $prodName
```

In Morel, I think it makes sense to intermingle 'in' and '=' in the same `from` expression. In XQuery if you want a `for` followed by a `let` followed by a `for` you have to create a new query (`for`), which if nothing else increases the indentation.

**Comment by julianhyde** [comment]
Fixed in 7fc889aced38e65fb582b39dec8bcbbdfd758e8d.
